dask_expr/_collection.py

```diff
@@ -145,6 +145,18 @@ def __getitem__(self, other):
             return new_collection(self.expr.__getitem__(other.expr))
         return new_collection(self.expr.__getitem__(other))
 
+    def persist(self, fuse=True, combine_similar=True, **kwargs):
+        out = self.optimize(combine_similar=combine_similar, fuse=fuse)
+        return DaskMethodsMixin.persist(
+            out, combine_similar=combine_similar, fuse=fuse, **kwargs
+        )
+
+    def compute(self, fuse=True, combine_similar=True, **kwargs):
+        out = self.optimize(combine_similar=combine_similar, fuse=fuse)
+        return DaskMethodsMixin.compute(
+            out, combine_similar=combine_similar, fuse=fuse, **kwargs
+        )
+
     def __dask_graph__(self):
         out = self.expr
         out = out.optimize(fuse=False)
```

Comment:
It feels "wrong" that we are always optimizing the task graph here, even if it is in service of a `compute` call with `combine_similar=False`.

I think we need to change the logic in `__dask_graph__`/`__dask_keys__`/etc to only do the minimal "lowering" needed to construct a task graph.  Right now we only have `lower_once`, so we probably need to introduce a `lower` method to repeat `lower_once` until the exression is "fully lowered". What do you think?